agnpy/emission_regions/blob.py

```diff
@@ -443,14 +458,25 @@ def gamma_recalculated_after_loss(gamma):
             bin_size_factor = 0.0001
             bins_width = gamma_bins_from * bin_size_factor
             gamma_bins_to = gamma_bins_from + bins_width
-            combined = np.column_stack((gamma_bins_from, gamma_bins_to)).ravel()
-            recalc = gamma_recalculated_after_loss(combined)
-            gamma_bins_from = recalc[0::2]
-            gamma_bins_to = recalc[1::2]
-            changed_bins_width = gamma_bins_to - gamma_bins_from
-            if np.any(changed_bins_width <= 0):
-                raise ValueError(
-                    "Energy loss formula returned too big value. Use shorter time ranges.")
+            gamma = interlace(gamma_bins_from, gamma_bins_to)
+            gamma_recalculated = gamma_recalculated_after_loss(gamma)
+            gamma_bins_from_recalc, gamma_bins_to_recalc = deinterlace(gamma_recalculated)
```

Comment:
I think lines 463-466 are only needed for Euler method, so they can go into the first if, or even better if you extract it all after if since it is common to the lines in the if for Heun

Reply:
as mentioned in the next comment, all these lines are needed to calculate the InterpolatedDistribution which is needed for the "second round" of the Heun method